### ISSUE TITLE
Problem when there are consecutive function calls on the right side of the assignment statement

### ISSUE THREAD

**Issue by Dongyang0810** [antlr/grammars-v4#3923]
`local a = GetClass(data).OtherFunc(self, ohterData)`
GetClass is a function that returns an object with a member function OtherFunc.
GetClass(data).OtherFunc(self, otherData) should be parsed as a tree structure where the root node is a function call, and the child node is also a function call.

The parsing result of GetClass(data).OtherFunc(self, ohterData) when parsing in a single line is inconsistent with the parsing result when placed on the right side of the assignment statement.

```
local a = GetClass(data).OtherFunc(self, ohterData)
GetClass(data).OtherFunc(self, ohterData)
```
![image](https://github.com/antlr/grammars-v4/assets/37564912/793918cc-8fd6-4a2e-9aae-92ffeb77abdf)

How can I resolve this issue?

**Comment by KvanTTT** [comment]
Please clarify the grammar you are talking about.

**Comment by KvanTTT** [comment]
Got it, I've seen `Lua` on the image.

**Comment by bkiers** [comment]
Instead of:

```
prefixexp
    : NAME ('[' exp ']' | '.' NAME)*
    | functioncall ('[' exp ']' | '.' NAME)*
    | '(' exp ')' ('[' exp ']' | '.' NAME)*
    ;
```

try this:


```
prefixexp
    : functioncall ('[' exp ']' | '.' NAME)*
    | NAME ('[' exp ']' | '.' NAME)*
    | '(' exp ')' ('[' exp ']' | '.' NAME)*
    ;
```

**Comment by bkiers** [comment]
This seems to be an exact duplicate of https://github.com/antlr/grammars-v4/issues/3885

**Comment by Dongyang0810** [comment]
> This seems to be an exact duplicate of https://github.com/antlr/grammars-v4/issues/3885
yes,I saw the question after I asked it.

**Comment by Dongyang0810** [comment]
> Instead of:
> 
> ```
> prefixexp
>     : NAME ('[' exp ']' | '.' NAME)*
>     | functioncall ('[' exp ']' | '.' NAME)*
>     | '(' exp ')' ('[' exp ']' | '.' NAME)*
>     ;
> ```
> 
> try this:
> 
> ```
> prefixexp
>     : functioncall ('[' exp ']' | '.' NAME)*
>     | NAME ('[' exp ']' | '.' NAME)*
>     | '(' exp ')' ('[' exp ']' | '.' NAME)*
>     ;
> ```

Yes, I have reordered the syntax rules like this, but I saw kaby76's comment: https://github.com/antlr/grammars-v4/issues/3885#issuecomment-1852090676, so this issue might require a different solution.

**Comment by kaby76** [comment]
> Yes, I have reordered the syntax rules like this, but I saw kaby76's comment: [#3885 (comment)](https://github.com/antlr/grammars-v4/issues/3885#issuecomment-1852090676), so this issue might require a different solution.

Of course, I may be wrong! Reordering may fix your problem, but I think the problem is that it doesn't solve the reported ambiguity.

**Comment by Dongyang0810** [comment]
> > Yes, I have reordered the syntax rules like this, but I saw kaby76's comment: [#3885 (comment)](https://github.com/antlr/grammars-v4/issues/3885#issuecomment-1852090676), so this issue might require a different solution.
> 
> Of course, I may be wrong! Reordering may fix your problem, but I think the problem is that it doesn't solve the reported ambiguity.

Is there a plan to fix this issue?

**Comment by bkiers** [comment]
> ...
> Is there a plan to fix this issue?

Feel free to submit a PR including some tests to validate it 😄
